arctic/store/version_store.py

```diff
@@ -247,19 +247,23 @@ def list_versions(self, symbol=None, snapshot=None, latest_only=False):
                 raise NoDataFoundException('No snapshot %s in library %s' % (snapshot, self._arctic_lib.get_name()))
 
         versions = []
+        snapshots = None
         for symbol in symbols:
             query['symbol'] = symbol
             seen_symbols = set()
             for version in self._versions.find(query, projection=['symbol', 'version', 'parent', 'metadata.deleted'], sort=[('version', -1)]):
                 if latest_only and version['symbol'] in seen_symbols:
                     continue
+                if snapshots is None:
+                    # populate only once and if is really going to be used
+                    snapshots = {ss.get('_id'): ss.get('name') for ss in self._snapshots.find()}
```

Comment:
I think this will always be hit - just always run above?

Reply:
sounds right, is an overkill to only avoid overhead on empty collections
moved it up